### PR TITLE
[CRN-1180] Add search bar and filter for working groups list page

### PR DESCRIPTION
**Pull request by ricardocarvalh0**
Jira ticket https://asaphub.atlassian.net/browse/CRN-1180

### CODE REVIEW

**Review by gabiayako — CHANGES_REQUESTED** [review]
looks nice! 🎉 just requesting changes because I think maybe we need to include user names in the search part, maybe it's blocked by members/leaders part though. Could you check with Tiff if this members part in the search is indeed a requirement? :) if so maybe it could be implemented already (maybe querying a non existent field does not give an error) or could be separated in another ticket after the merge with leaders/members

File: apps/crn-frontend/src/network/users/UserList.tsx
Comment (ricardocarvalh0) [reply apps/crn-frontend/src/network/users/UserList.tsx:42]: I guess it shouldn't apply the filters for users since it's prefetching the list to show it faster on a different tab (with diff filters). Groups and Teams are doing the same.

File: apps/crn-server/src/controllers/working-groups.ts
Comment (ricardocarvalh0) [reply apps/crn-server/src/controllers/working-groups.ts:20]: Hmm maybe `workingGroupFilter` since it can filter by additional columns in the future.

File: apps/crn-server/src/data-providers/working-groups.data-provider.ts
Comment (ricardocarvalh0) [reply apps/crn-server/src/data-providers/working-groups.data-provider.ts:65]: You are right, thank you!

**Comment by ricardocarvalh0** [comment]
> 

Confirmed with Tiffany the member's search was removed from the ticket scope.

File: apps/crn-frontend/src/network/users/UserList.tsx
Comment (gabiayako) [reply apps/crn-frontend/src/network/users/UserList.tsx:42]: shouldn't all usePrefetch have filter as `new Set()` then? In the TeamList I saw that I was the one who changed from `filters: new Set()` to `filters` 🙈 

originally they were set to `filters: new Set()` https://github.com/yldio/asap-hub/pull/795/files although we keep the search term when we change tabs. Like If I type `test` in the search bar when `Teams` is selected, if I click in the `Interest Groups` tab then it still filters by `test`. So I'm not sure which one it should be, but for me it would make sense for them all to be the same 🤔

Comment (ricardocarvalh0) [reply apps/crn-frontend/src/network/users/UserList.tsx:42]: I see, it seems UserList is the only one that uses `new Set()`. I think it's fine to change it then

**Review by gabiayako — APPROVED** [review]
Nice! 🎉🎉🎉

File: apps/crn-frontend/src/network/teams/TeamList.tsx
Comment (gabiayako) [inline apps/crn-frontend/src/network/teams/TeamList.tsx:35]: I've just asked @kmaid about this, we should not preserve the filters.

About my past comment, I was making a confusion with `searchQuery` and `filters` which are not the same.

Since the filters change from one entity to another, we shouldn't preserve then, so actually we should have `filters: new Set()`.

Sorry about the confusion 😅

**Review by gabiayako — CHANGES_REQUESTED** [review]

**Pull request merged**
